x-pack/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -43,168 +49,184 @@ function ActionMenuButton({ onClick }: { onClick: () => void }) {
+      path: `/link-to/container-detail/${containerId}`,
+      query: infraMetricsQuery
+    },
+    {
+      icon: 'infraApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showHostMetricsLinkLabel',
+        { defaultMessage: 'Show host metrics' }
+      ),
+      condition: hostName,
+      path: `/link-to/host-detail/${hostName}`,
+      query: infraMetricsQuery
+    }
+  ].map(({ icon, label, condition, path, query }, index) =>
+    !!condition
+      ? {
+          icon,
+          key: `infra-link-${index}`,
+          child: (
+            <InfraLink path={path} query={query}>
+              {label}
+            </InfraLink>
+          )
+        }
+      : null
+  );
 
-    const items = [
-      ...this.getInfraActions(),
-      <EuiContextMenuItem icon="discoverApp" key="discover-transaction">
-        <EuiFlexGroup gutterSize="s">
-          <EuiFlexItem>
-            <DiscoverTransactionLink transaction={transaction}>
-              {i18n.translate(
-                'xpack.apm.transactionActionMenu.viewSampleDocumentLinkLabel',
-                {
-                  defaultMessage: 'View sample document'
-                }
-              )}
-            </DiscoverTransactionLink>
-          </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <EuiIcon type="popout" />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiContextMenuItem>
-    ];
+  const uptimeLink = url.format({
+    pathname: chrome.addBasePath('/app/uptime'),
+    hash: `/?${fromQuery(
+      pick(
+        {
+          dateRangeStart: urlParams.rangeFrom,
+          dateRangeEnd: urlParams.rangeTo,
+          search: urlParams.kuery
+        },
+        (val: string) => !!val
+      )
+    )}`
+  });
 
-    return (
-      <EuiPopover
-        id="transactionActionMenu"
-        button={<ActionMenuButton onClick={this.toggle} />}
-        isOpen={this.state.isOpen}
-        closePopover={this.close}
-        anchorPosition="downRight"
-        panelPaddingSize="none"
-      >
-        <EuiContextMenuPanel
-          items={items}
-          title={i18n.translate(
-            'xpack.apm.transactionActionMenu.actionsLabel',
-            { defaultMessage: 'Actions' }
+  const menuItems = removeEmptyItems([
+    ...infraItems,
+    {
+      icon: 'discoverApp',
+      key: 'discover-transaction',
+      child: (
+        <DiscoverTransactionLink transaction={transaction}>
+          {i18n.translate(
+            'xpack.apm.transactionActionMenu.viewSampleDocumentLinkLabel',
+            {
+              defaultMessage: 'View sample document'
+            }
           )}
-        />
-      </EuiPopover>
-    );
-  }
-}
+        </DiscoverTransactionLink>
+      )
+    },
+    {
+      icon: 'uptimeApp',
+      key: 'uptime',
+      child: (
+        <EuiLink href={uptimeLink}>
+          {i18n.translate('xpack.apm.transactionActionMenu.viewInUptime', {
+            defaultMessage: 'View in Uptime'
```

Comment:
Change default message to: "View monitor status"

Reply:
@dgieselaar You're too fast 😄

Reply:
@formgeist haha 🚀🚀🚀